src/analysis/retail/evoker/devastation/modules/AplCheck/index.tsx

```diff
@@ -20,64 +21,421 @@ const hasEssenceRequirement = (resources: number) => {
 };
 
 const COMMON_TOP: Rule[] = [
+  // Spend Snapfire procs ASAP
+  {
+    spell: TALENTS.FIRESTORM_TALENT,
+    condition: cnd.buffPresent(SPELLS.SNAPFIRE_BUFF),
+  },
+  {
+    spell: TALENTS.SHATTERING_STAR_TALENT,
+    condition: cnd.describe(cnd.optionalRule(cnd.spellAvailable(TALENTS.DRAGONRAGE_TALENT)), () => (
+      <>
+        you are about to cast <SpellLink spell={TALENTS.DRAGONRAGE_TALENT} /> <strong>(AoE)</strong>
+      </>
+    )),
+  },
+  {
+    spell: TALENTS.DRAGONRAGE_TALENT,
+    condition: cnd.describe(
+      cnd.optionalRule(
+        cnd.or(
+          cnd.targetsHit(
+            { atLeast: 3 },
+            { lookahead: 3000, targetType: EventType.Damage, targetSpell: SPELLS.PYRE },
+          ),
+          cnd.and(
+            cnd.spellCooldownRemaining(SPELLS.FIRE_BREATH, { atMost: 4000 }),
+            cnd.spellCooldownRemaining(SPELLS.ETERNITY_SURGE, { atMost: 10000 }),
+          ),
+        ),
+      ),
+      () => (
+        <>
+          it would hit at least 3 targets or <SpellLink spell={SPELLS.FIRE_BREATH} /> will be off
+          cooldown in less than 4 seconds and <SpellLink spell={SPELLS.ETERNITY_SURGE} /> has no
+          more than 10 seconds remaining on cooldown
+        </>
+      ),
+    ),
+  },
+  // With Event Horizon talented, use Eternity Surge before Fire Breath during DR
+  {
+    spell: SPELLS.ETERNITY_SURGE,
+    condition: cnd.and(
+      cnd.hasTalent(TALENTS.EVENT_HORIZON_TALENT),
+      cnd.buffPresent(TALENTS.DRAGONRAGE_TALENT),
+    ),
+  },
+  {
+    spell: SPELLS.ETERNITY_SURGE_FONT,
+    condition: cnd.and(
+      cnd.hasTalent(TALENTS.EVENT_HORIZON_TALENT),
+      cnd.buffPresent(TALENTS.DRAGONRAGE_TALENT),
```

Comment:
needs another `cnd.hasTalent(TALENTS.FONT_OF_MAGIC)` and above the inverse?

Reply:
The rules only apply if the spell is able to be cast - since you cant cast font ES if you don't have the talent and vice versa, this will not be a problem, since it will only show the talented spell. This is the original solution used to deal with font, I just kept it for now.

Reply:
or i guess it just doesnt fire if the spells were never used? might be fine